src/backend/marsha/core/xapi.py

```diff
@@ -0,0 +1,106 @@
+"""XAPI module."""
+from datetime import datetime, timezone
+import re
+import uuid
+
+import requests
+
+
+class XAPI:
+    """The LTI object abstracts an LTI launch request.
+
+    It provides properties and methods to inspect the launch request.
+    """
+
+    def __init__(self, video, statement, lti_user):
+        """Initialize the XAPI module.
+
+        Parameters
+        ----------
+        video : Type[.models/videos]
+            The video object used in the xAPI statement
+
+        statement : dictionary
+            Statement containing base information to send to the LRS
+            An example of expected statement:
+            {
+                "verb": {
+                    "id": "http://adlnet.gov/expapi/verbs/initialized",
+                    "display": {
+                        "en-US": "initialized"
+                    }
+                },
+                "context": {
+                    "extensions": {
+                        "https://w3id.org/xapi/video/extensions/volume": 1,
+                        "https://w3id.org/xapi/video/extensions/video-playback-size": "640x264",
+                    }
+                }
+            }
+
+        lti_user : .lti.LTIUser
+            Object representing data stored in the JWT Token and related to the user authenticated
+            with LTI
+
+        """
+        self.video = video
+        self.statement = statement
+        self.lti_user = lti_user
+
+    def send(self, url, basic_auth_token):
+        """Send the statement to a LRS.
+
+        Parameters
+        ----------
+        url : string
+            The LRS endpoint to fetch
+
+        basic_auth_token : string
+            The basic_auth token used to authenticate on the LRS
+
+        """
+        self._enrich_statement()
+
+        headers = {
+            "Authorization": basic_auth_token,
+            "Content-Type": "application/json",
+            "X-Experience-API-Version": "1.0.1",
+        }
+
+        response = requests.post(
+            "{url}/statements".format(url=url), json=self.statement, headers=headers
```

Comment:
To me, we should provide a statement endpoint instead of hard-coding the endpoint given an url. Some other LRS may implement different endpoints right?

Reply:
Yes but if we want to implement other LRS we should maybe refactor this class and delegate the responsability to query the LRS to a dedicated class ? One class per LRS and a factory to choose the good one ?

Reply:
if it's really a standard as I understand it only the url and credentials should vary?

Reply:
https://github.com/adlnet/xAPI-Spec/blob/xAPI-1.0.3/xAPI-Communication.md#10-requests Yes I see nothing about the endpoint and how to connect